Orange/projection/pca.py

```diff
@@ -137,16 +153,18 @@ def partial_fit(self, data):
         return self
 
 
-class Projector:
-    def __init__(self, projection, feature):
+class Projector(ComputeValueWCommon):
+    """Transform into a given PCA component."""
+
+    def __init__(self, projection, feature, commonfn):
         self.projection = projection
         self.feature = feature
         self.transformed = None
+        self.commonfn = commonfn #domain conversion to original data set,
+                                 #transform to PCA space
 
-    def __call__(self, data):
-        if data.domain != self.projection.pre_domain:
-            data = data.from_table(self.projection.pre_domain, data)
-        return self.projection.transform(data.X)[:, self.feature]
```

Comment:
My line of reasoning. I would like to clear the cache after transformation. The only thing that knows when transformation ends is the user. Therefore the user has to be aware of it.

An alternative solution would be to call something like clear_cache() for all the variables after transformation, but this has the same awareness-drawback while being less direct. Also, theoretically, such caches would have problems with concurrent transforms...